DEV-INTRO.md

```diff
@@ -1,59 +1,187 @@
+### First time setup
 
-* Fastest way to get started
-* see PEcAn demo ...
+The steps in this section only need to be done the fist time you start working with the stack in docker. After this is done you can skip these steps. You can find more detail about the docker commands in the [pecan documentation](https://pecanproject.github.io/pecan-documentation/master/docker-index.html).
 
-### Installing from source
+Before doing anything it is recommended to make sure you have the lastest docker images ready. You can do a `docker-compose pull` to get the latest images.
 
-#### From GitHub
+* setup .env file
+* create folders to hold the data
+* load the postgresql database
+* load some test data
+* copy all R packages (optional but recommended) 
+* setup for web folder development (optional)
 
+#### .env file
+
+You can copy the [`env.example`](docker/env.example) file as .env in your pecan folder. The variables we want to modify are:
+
+* `COMPOSE_PROJECT_NAME` set this to pecan, the prefix for all containers
+* `PECAN_VERSION` set this to develop, the docker image we start with
+
+At the end you should see the following if you run the following command `egrep -v '^(#|$)' .env`
+
+```
+COMPOSE_PROJECT_NAME=pecan
+PECAN_VERSION=develop
+```
+
+#### folders
+
+Next we will create the folders that will hold all the data for the docker containers using: `mkdir -p volumes/{lib,pecan,portainer,postgres,rabbitmq,traefik}`. The `volumes` folder will be ignored by git. You can create these at any location, however you will need to update the `docker-compose.dev.yml` file. The subfolders are used for the following:
+
+- **lib** holds all the R packages for the specific version of PEcAn and R. This folder will be shared amongst all other containers, and will contain the compiled PEcAn code.
+- **pecan** this holds all the data, such as workflows and any downloaded data.
+- **portainer** if you enabled the portainer service this folder is used to hold persistent data for this service
+- **postgres** holds the actual database data. If you want to backup the database, you can stop the postgres container, zip up the folder.
+- **rabbitmq** holds persistent information of the message broker (rabbitmq). 
+- **traefik** holds persisent data for the web proxy, that directs incoming traffic to the correct container.
+
+These folders will hold all the persistent data for each of the respective containers and can grow. For example the postgres database is multiple GB. The pecan folder will hold all data produced by the workflows, including any downloaded data, and can grow to many giga bytes.
+
+#### postgresql database
+
+First we bring up postgresql (we will start RabbitMQ as well since it takes some time to start): `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`. This will start postgresql and rabbitmq. We need to wait for a few minutes (you can look at the logs using `docker-compose logs postgres`) to see if it is ready.
```

Comment:
If they copied `docker-compose.dev.yml` to `docker-compose.override.yml` and have them both in the directory, this command (`docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`) will run just fine, but do we want to stick to `docker-compose up -d postgres rabbitmq`

Reply:
I agree with @istfer on sticking with using `override.yml` throughout. Now that I'm running through the "First time setup" again to see if I can get that Wisconsin data, I'm having to switch back and forth between `dev.yml` and `override.yml`, which is confusing. 

I think it makes sense to have the file named `docker-compose.dev.yml` in the repo, but then at the beginning of these instructions (not sure exactly where) to have the command `mv docker-compose.dev.yml docker-compose.override.yml`. Then the rest of the documentation should be updated accordingly, which it looks like Istem has already been doing.

Reply:
yes, I forgot the update the documentation to fix this. Now you can just use `docker-compose up -d` no need to specify the different yml files.